specs/Menu.md

```diff
@@ -1,731 +1,883 @@
+    key: 'editorials',
+    children: (El, props) => <El>{props.key}</El>
+  },
+]
+
+<Menu defaultActiveIndex={0} items={items} />
+
+// v0 children API custom render
+<Menu>
+  <Menu.Item index={0}>
+    <Menu.ItemContent>Editorials</Menu.ItemContent>
+  </Menu.Item>
+  <Menu.Item index={1}>
+    CustomContent
+  </Menu.Item>
+  {/*Not recommended but definitely possible*/}
+  <div>custom item</div>
+</Menu>
+```
 
-Maybe should be named SubmenuContext?
-Should submenu have its own context?
+## Variants
 
-```TSX
-function() {
-return (
-    <OpenableContext.Provider value={submenuState}>
-        {props.children}
-    </OpenableContext.Provider>);
-}
-```
+### Nested menus
 
-#### SubmenuList
+A `Menu` should be able to trigger an additional instance of itself as a part of one or more of its options. The nested `Menu` component should have the same functional capabilities as the root `Menu` component.
 
-```TSX
-function() {
-    const openableContext = React.useContext(OpenableContext);
-    return (openableContext.open &&
-    <Slots.Root {...menuProps}>
-        {props.children}
-    </Slots.Root>);
-}
-```
+The actions that trigger the the nested `Menu` should be be consistent with the actions that can trigger any root `Menu` from a similar UI control.
 
-#### SubmenuItem
+We advise that no more than two nested `Menu` components be used, but this spec does not functionally apply that constrain to the implementation of the `Menu` component.
 
-```TSX
-function() {
-    const openableContext = React.useContext(OpenableContext);
-    return (
-    <Slots.Root {...menuItemProps} onClick={openableContext.toggleOpen}>
-        {props.children}
-    </Slots.Root>);
-}
+### Selection state
+
+A `Menu` should be able to track and represent the selection state of all or some of its options if required.
+
+When an options is associated with a selection state. The `Menu`, either root or nested, should control its dismiss behavior accordingly based on configuration.
+
+### Sections
+
+A `Menu` can be partitioned into sections using visible dividers in its list of options. Each section can contain a heading title that announces or briefly describes the options in the particular section
+
+### Secondary label
+
+An option of a `Menu` component should be able to declare additional secondary label that can provide additional context describing the option or its usage.
+
+For example a secondary label can be a label that shows a keyboard shortcut that will perform an equivalent action of the option of the `Menu` component.
+
+### Split option with nesting
+
+An option of a `Menu` component can trigger a nested `Menu` component and also perform its default action by splitting the option into two interactable areas that handle each action separately.
+
+### Disabled option(s)
+
+All options in a `Menu` component can be disabled and should provide a visible indication for this purpose. User interaction should be defined for disabled options
+
+### Scrollable
```

Comment:
Do we need to provide any kind of UI Virtualization hooks for scrolling long lists of menu items? One example, font picking scenarios where a user could have 300+ fonts on the system.

Reply:
That's a very good question, I'd love to schedule some time with @YuanboXue-Amber to talk about her recent efforts in virtualizing the v0 Tree with a hooks refactor and and try to leverage her experiences in how this proposed API could do in a similar situation

Reply:
Also from what I see neither v0 nor v7 currently support the concept of a virtualized menu, it's rather list that should be virtualized

Reply:
After and audit of some virtualization libraries and a talk with @YuanboXue-Amber who did virtualization recently for the v0 `Tree`, it seems a children based API will play nicely. Since this is also not supported in either v7 or v0, let's open that can of worms once we get there